cloud/amazon/lambda.py

```diff
@@ -0,0 +1,395 @@
+    required: false
+    default: null
+  timeout:
+    description:
+      - The function execution time at which Lambda should terminate the function.
+    required: false
+    default: 3
+  memory_size:
+    description:
+      - The amount of memory, in MB, your Lambda function is given
+    required: false
+    default: 128
+notes:
+  - 'Currently this module only supports uploaded code via S3'
+author:
+    - 'Steyn Huizinga (@steynovich)'
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Create Lambda functions
+tasks:
+- name: looped creation
+  lambda:
+    name: '{{ item.name }}'
+    state: present
+    path: '{{ item.path }}'
+    runtime: 'python2.7'
+    role_arn: 'arn:aws:iam::987654321012:role/lambda_basic_execution'
+    handler: 'hello_python.my_handler'
+  with_items:
+    - { name: HelloWorld, path: 'hello-code.zip' }
+    - { name: ByeBye, path: 'bye-code.zip' }
+
+# Basic Lambda function deletion
+tasks:
+- name: Delete Lambda functions HelloWorld and ByeBye
+  lambda:
+    name: '{{ item }}'
+    state: absent
+  with_items:
+    - HelloWorld
+    - ByeBye
+'''
+
+RETURN = '''
+output:
+  description: the data returned by create_function in boto3
+  returned: success
+  type: dict
+  sample: 
+    {
+        'FunctionName': 'string',
+        'FunctionArn': 'string',
+        'Runtime': 'nodejs',
+        'Role': 'string',
+        'Handler': 'string',
+        'CodeSize': 123,
+        'Description': 'string',
+        'Timeout': 123,
+        'MemorySize': 123,
+        'LastModified': 'string',
+        'CodeSha256': 'string',
+        'Version': 'string',
+    }
+'''
+
+try:
+    import botocore
+    HAS_BOTOCORE = True
+except ImportError:
+    HAS_BOTOCORE = False
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+try:
+    import binascii
+    HAS_BINASCII = True
+except ImportError:
+    HAS_BINASCII = False
+
+try:
+    import base64
+    HAS_BASE64 = True
+except ImportError:
+    HAS_BASE64 = False
+
+try:
+    import hashlib
+    HAS_HASHLIB = True
+except ImportError:
+    HAS_HASHLIB = False
+
+
+def get_current_function(connection, function_name):
+    try:
+        return connection.get_function(FunctionName=function_name)
+    except botocore.exceptions.ClientError as e:
+        return False
+
+
+def sha256sum(filename):
+    hasher = hashlib.sha256()
+    with open(filename, 'rb') as f:
+        hasher.update(f.read())
+
+    code_hash = hasher.digest()
+    code_b64 = base64.b64encode(code_hash)
+    hex_digest = code_b64.decode('utf-8')
+
+    return hex_digest
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present','absent']),
+        runtime=dict(type='str', choices=['nodejs','nodejs4.3','java8','python2.7'], default=None),
```

Comment:
I recommend against requiring only those choices. I'd rather rely on catching `ClientError` later, since AWS is likely to update the supported languages and it would be unfortunate to have Ansible prevent folks from using them.